### PR TITLE
Bridge tx history

### PR DESCRIPTION
**Pull request by r-czajkowski**
Depends on: #321
Closes: #289 

This PR adds support for fetching bridge transaction history in order from the newest to the oldest revealed deposit.

obraz

### CODE REVIEW

File: src/threshold-ts/tbtc/index.ts [hunk 1]
```diff
@@ -223,4 +263,105 @@ export class TBTC implements ITBTC {
...
+      )
+
+      return {
+        amount: "0",
+        walletPublicKeyHash: deposit.args?.walletPubKeyHash,
+        fundingTxHash,
+        fundingOutputIndex,
+        depositKey,
+        txHash: deposit.transactionHash,
+      }
```
Comment (michalsmiarowski) [inline src/threshold-ts/tbtc/index.ts:327]: I think we should print the tx history from the newest (at the top) to the oldest. Right now it's displayed from the oldest to the newest.

It looks like it gets the `DepositRevealed` events sorted atuomatically from the oldest to newest so I think we sohuld be good by just reversing the order of the events.

The other way would be to get `refundLocktime` from the script and sort it by this value. The problematic thing might be that this value is a 4-byte little-endian un-prefixed hex string so we would probably have to change it to decimal and then sort.

File: src/threshold-ts/tbtc/index.ts [hunk 2]
```diff
@@ -223,4 +263,105 @@ export class TBTC implements ITBTC {
...
+          interface: this._bridgeContract.interface,
+          method: "deposits",
+          args: [depositKey],
+        }))
+    )
+
+    return parsedDeposits.map((deposit, index) => {
+      const amount = revealedDeposits[index][0]?.amount?.toString()
+      return { ...deposit, amount }
+    })
```
Comment (michalsmiarowski) [inline src/threshold-ts/tbtc/index.ts:344]: Is this really needed? 🤔 In 76f2420ae2e974f40c4c5571d59f7170cc82977d commit message you wrote:
```
`DepositRevealed` event doesn't include `amount` param so we need to
call `Brodge.deposits` function to get an `amount` for a given deposit
key.
```
but this doesn't seem to be true. `DepositRevealed` event does include amount param. Unless you meant something else?

File: src/pages/tBTC/Bridge/index.tsx [hunk 3]
```diff
@@ -46,7 +62,7 @@ const TBTCBridge: PageComponent = (props) => {
       </Box>
       <Box gridArea="aside">
         <TbtcBalanceCard gridArea="balance-card" mb="5" />
-        <TransactionHistory gridArea="tx-history" mb="5" />
+        <TransactionHistory data={bridgeTransactionHistory} />
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/index.tsx:65]: We should update the data after the reveal is done so the user won't have to refresh the page to get the current tx history.

We listen for the `DepositRevealed` event here:
https://github.com/threshold-network/token-dashboard/blob/6c2941e062626160f1d331403ca0297d78de4a20/src/hooks/tbtc/useSubsribeToDepositRevealedEvent.ts#L7
so I think we could update the data there.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Overall looking good, but added some comments to look at  👀

File: src/threshold-ts/tbtc/index.ts (hunk 2 above)
Comment (r-czajkowski) [reply src/threshold-ts/tbtc/index.ts:344]: You are right! We can get an amount from `DepositRevealed` event- it's `fundingOutputAmount` param. I missed this param. Thanks!

Comment (r-czajkowski) [reply src/threshold-ts/tbtc/index.ts]: 453961d

File: src/threshold-ts/tbtc/index.ts (hunk 1 above)
Comment (r-czajkowski) [reply src/threshold-ts/tbtc/index.ts]: 453961d

File: src/pages/tBTC/Bridge/TransactionHistory/index.tsx [hunk 4]
```diff
@@ -94,27 +66,38 @@ export const TransactionHistory: FC<ComponentProps<typeof Card>> = ({
...
             </LabelSm>
           </Tr>
         </Thead>
         <BodySm as={Tbody}>
           {isHistoryEmpty ? (
             <EmptyHistoryTableBody />
           ) : (
-            history.map((tx, index) => (
-              <Tr key={index}>
+            data.map((_) => (
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/TransactionHistory/index.tsx:98]: 41c4d44

File: src/pages/tBTC/Bridge/index.tsx (hunk 3 above)
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/index.tsx]: 12cc6ff

**Review by michalsmiarowski — APPROVED** [review]
LGTM. We will address the "Expand view" button in a separate PR.

**Pull request merged**